market-generator/src/main/java/com/market/banica/generator/service/PersistSchedulerImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.market.banica.generator.service;
+
+import com.market.banica.generator.model.MarketTick;
+import com.market.banica.generator.service.task.SnapshotPersistenceTask;
+import com.market.banica.generator.util.SnapshotPersistence;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jmx.export.annotation.ManagedOperation;
+import org.springframework.jmx.export.annotation.ManagedResource;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.Timer;
+import java.util.concurrent.TimeUnit;
+
+@Component
+@ManagedResource
+public class PersistSchedulerImpl {
```

Comment:
Shouldn't this class implement the PersistScheduler interface?

Reply:
Fixed

Reply:
Fixed